------ hibernate-core/src/main/java/org/hibernate/mapping/ForeignKey.java ------
index 6732f10..e8b1a4c 100644
@@ -58,7 +58,7 @@ public class ForeignKey extends Constraint {
 			i++;
 		}
 		String result = dialect.getAddForeignKeyConstraintString(
-			constraintName, cols, referencedTable.getQualifiedName(dialect, defaultCatalog, defaultSchema), refcols, isReferenceToPrimaryKey()
+			dialect.quote(constraintName), cols, referencedTable.getQualifiedName(dialect, defaultCatalog, defaultSchema), refcols, isReferenceToPrimaryKey()
 		);
 		return cascadeDeleteEnabled && dialect.supportsCascadeDelete() ? 
 			result + " on delete cascade" : 